main.go

```diff
@@ -0,0 +1,123 @@
+package authenticatedblockstore
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"os"
+	"strings"
+	"time"
+)
+
+// This is a test file for testing the package.
+// It is not intended to be part of the package.
+
+var (
+	logFileNames   []string
+	logFileHandles []*os.File
+)
+
+type logFileNode struct {
```

Comment:
Each of these fields ought to be commented and described in detail, since this will be essential for future maintainers trying to understand the implementation / algorithm.

Reply:
Sure, the main.go is removed, as it is for temporary testing only. 
I will have the committed files better documented.